src/WebJobs.Script/Config/TransmissionStatusHandler.cs

```diff
@@ -0,0 +1,147 @@
+            if (args.Response?.StatusCode != 200 || _source.IsEnabled(ScriptConstants.HostDiagnosticSourceDebugEventNamePrefix))
+            {
+                _source.Write("TransmissionStatus", FormattedLog(sender, args));
+            }
+        }
+
+        internal string FormattedLog(object sender, TransmissionStatusEventArgs args)
+        {
+            var transmission = sender as Transmission;
+            var items = transmission?.TelemetryItems.GroupBy(n => n.GetEnvelopeName())
+                        .Select(n => new TelemetryItem
+                        {
+                            Type = n.Key,
+                            Count = n.Count()
+                        });
+
+            IngestionServiceResponse response = null;
+            if (!string.IsNullOrWhiteSpace(args?.Response?.Content))
+            {
+                response = JsonSerializer.Deserialize(
+                    args.Response.Content, IngestionServiceResponseContext.Default.IngestionServiceResponse);
+            }
+
+            string topErrorMessage = null;
+            int? topStatusCode = null;
+            if (response?.Errors != null && response.Errors.Length > 0)
+            {
+                topErrorMessage = response.Errors[0].Message;
+                topStatusCode = response?.Errors[0].StatusCode;
+            }
+
+            var log = new LogMessage
+            {
+                StatusCode = args?.Response?.StatusCode,
+                StatusDescription = args?.Response?.StatusDescription,
+                Id = transmission?.Id,
+                Items = items,
+                Received = response?.ItemsReceived,
+                Accepted = response?.ItemsAccepted,
+                ErrorMessage = topErrorMessage,
+                ErrorCode = topStatusCode,
+                ResponseTimeInMs = args?.ResponseDurationInMs,
+                RetryAfterHeader = args?.Response?.RetryAfterHeader
+            };
+            return JsonSerializer.Serialize(log, LogMessageContext.Default.LogMessage);
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        public void Dispose()
+        {
+            _source.Dispose();
+        }
+    }
+
+    [JsonSerializable(typeof(IngestionServiceResponse), GenerationMode = JsonSourceGenerationMode.Metadata)]
+    internal partial class IngestionServiceResponseContext : JsonSerializerContext
+    {
+    }
+
+    [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase, WriteIndented = true, GenerationMode = JsonSourceGenerationMode.Serialization, DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull)]
+    [JsonSerializable(typeof(LogMessage))]
+    internal partial class LogMessageContext : JsonSerializerContext
+    {
+    }
+
+    internal class IngestionServiceResponse
+    {
+        [JsonPropertyName("itemsReceived")]
+        public int? ItemsReceived { get; set; }
+
+        [JsonPropertyName("itemsAccepted")]
```

Comment:
Do we need all these `JsonPropertyName` attributes if we're setting CamelCase on the serializer itself?

Reply:
IngestionServiceResponse is only used for Deserialize.